scripts/alert_emails/send_emails.ts

```diff
@@ -162,73 +168,95 @@ async function setLastSnapshotNumber(
-              if (
-                err.Code === CM_INVALID_EMAIL_ERROR_CODE &&
-                err.Message === CM_INVALID_EMAIL_MESSAGE
-              ) {
-                invalidEmailCount += 1;
-                const currentData = (
-                  await db.collection('alerts-subscriptions').doc(doc.id).get()
-                ).data();
-                if (currentData) {
-                  await db
-                    .collection('invalid-alerts-subscriptions')
-                    .doc(doc.id)
-                    .set(currentData);
-                  await db
-                    .collection('alerts-subscriptions')
-                    .doc(doc.id)
-                    .delete();
-                }
-              } else {
-                errorCount += 1;
-              }
-            });
-        }
-      });
+      .get();
+    return querySnapshot.docs.map(emailDoc => emailDoc.id);
+  };
+
+  async function onInvalidEmail(email: string) {
+    invalidEmailCount += 1;
+    const querySnapshot = await db
+      .collection('alerts-subscriptions')
+      .doc(email)
+      .get();
+    const currentData = querySnapshot.data();
+    if (currentData) {
+      await db
+        .collection('invalid-alerts-subscriptions')
+        .doc(email)
+        .set(currentData);
+      await db.collection('alerts-subscriptions').doc(email).delete();
+    }
+  }
+
+  async function onEmailSent(email: string, fips: string) {
+    emailSent += 1;
+    uniqueEmailAddress[email] = (uniqueEmailAddress[email] || 0) + 1;
+    locationsWithEmails[fips] = (locationsWithEmails[fips] || 0) + 1;
+    // comment out this code if you are developing locally
+    return db
+      .collection(`snapshots/${currentSnapshot}/locations/${fips}/emails/`)
+      .doc(email)
+      .set({ sentAt: admin.firestore.FieldValue.serverTimestamp() });
+  }
+
+  async function sendAlertEmail(email: string, fips: string) {
+    const locationAlert = locationsWithAlerts[fips];
+    const sendData = generateSendData(email, locationAlert);
+
+    try {
+      const response = await sendEmail(api, sendData, dryRun);
+      await onEmailSent(email, fips);
+      return response;
+    } catch (err) {
+      if (isInvalidEmail(err)) {
+        await onInvalidEmail(email);
+      } else {
+        errorCount += 1;
+      }
+    }
   }
+
+  async function sendEmailBatch(emails: string[], fips: string) {
+    if (emails.length === 0) {
+      return;
+    }
+
+    const [firstEmail, ...remainingEmails] = emails;
+    const pilotReq = await sendAlertEmail(firstEmail, fips);
+
+    const rateLimitRemaining = parseInt(pilotReq.headers['x-ratelimit-limit']);
+    const rateLimitResetSec =
+      parseInt(pilotReq.headers['x-ratelimit-reset']) + 15;
+
+    if (rateLimitRemaining < remainingEmails.length + 10) {
+      console.log(
+        `Rate limit remaining: ${rateLimitRemaining}. Sleeping ${rateLimitResetSec} seconds.`,
+      );
+      await delay(rateLimitResetSec * 1000);
+    }
```

Comment:
I think we can drop the `firstEmail` special handling and just move the rate limit handling inside of sendAlertEmail() so that it works similar to before:

1. Try sending the email.
2. Check the response headers.
3. If we hit (or got close to) the rate limit, do the delay, and then just call sendAlertEmail() again to retry.

This means we might have more than one email from the batch hit the rate limit handling and have to sleep / retry.  But I think that's fine.  They'll all do it in parallel, so it doesn't actually add any extra time.  

The one messy bit is it's kinda' ugly if they all log a message about hitting the rate limit.  But since I think this will be rare we can either live with that (💯  okay by me) or else add some sort of flag so that only the first one that hits the rate limit logs the message.